gpkit/geometric_program.py

```diff
@@ -177,6 +183,55 @@ def solve(self, solver=None, verbosity=1, *args, **kwargs):
         else:
             return result
 
+    def check_solution(self, sol, tol=1e-5):
+        """Run a series of checks to mathematically confirm sol solves this GP
+
+        Arguments
+        ---------
+        sol: dict
+            solution dict, same format as return type of solve()
+
+        Raises
+        ------
+        RuntimeWarning, if any problems are found
+        """
+        def _almost_equal(a, b):
+            "local almost equal test"
+            return abs(np.log(a/b)) < tol
+        cost = sol["cost"]
+        primal_sol = np.log(sol["variables"].values())
+        nu = sol["sensitivities"]["monomials"]
+        la = sol["sensitivities"]["posynomials"]
+        A = self.A.tocsr()
```

Comment:
So, to clarify: 

1. are we dependent on scipy?
2. Does this commit change the answer to #1?